wandb/sdk/artifacts/artifact.py

```diff
@@ -1814,7 +1800,7 @@ def _download_entry(
                         # Handled by core
                         continue
                     entry._download_url = edge["node"]["directUrl"]
-                    if self.should_download_entry(entry, prefix=path_prefix):
+                    if not path_prefix or entry.path.startswith(str(path_prefix)):
```

Comment:
nit(optional):
```suggestion
                    if (not path_prefix) or entry.path.startswith(str(path_prefix)):
```

(not a bug, I just prefer it for readability whenever `not` is in front of an `and/or` clause.  definitely subjective/optional, though)